test/unit/date/parse.js

```diff
@@ -471,6 +471,12 @@ QUnit.test( "should parse milliseconds in a day (A+)", function( assert ) {
 	assertParse( assert, "6330736900", "AAAAA", cldr, date1 );
 });
 
+QUnit.test( "should allow strict parsing of the date", function( assert ) {
+	date1 = new Date(2001,0,1);
+	assertParse( assert, "1/1/01", "M/d/yy", cldr, date1, true );
+	assertParse( assert, "1/1/2001", "M/d/yy", cldr, null, true );
+});
```

Comment:
> My intention was to avoid 2016 year being parsed as 2020 year when yy matches first two digits of 20116.

I understand, but supporting `strict` should work for other fields as well.

> It appears to me that M will match both padded and not padded month... By that I mean it appears to be correct to parse `01` under `M` pattern

Formatting `January` with `M` in `en` will always return `"1"`, not `"01"`, so I think `01` is incorrect. Although, `12` for `December` would be correct.

Some investigation would be good. If it's a fix we could include as part of this PR I'd prefer.